base/datafmt.jl

```diff
@@ -119,6 +119,7 @@ readdlm_auto(input::Vector{UInt8}, dlm::Char, T::Type, eol::Char, auto::Bool; op
 readdlm_auto(input::IO, dlm::Char, T::Type, eol::Char, auto::Bool; opts...) =
     readdlm_string(read(input, String), dlm, T, eol, auto, val_opts(opts))
 function readdlm_auto(input::AbstractString, dlm::Char, T::Type, eol::Char, auto::Bool; opts...)
+    isfile(input) || error("Cannot read \'$input\': not a file")
```

Comment:
The issue was originally filed b/c someone tried to `readdlm` an existing directory. None of the Libc errors seem to really fit.